extensions/wikia/AdEngine/js/template/porvata.js

```diff
@@ -99,22 +97,26 @@ define('ext.wikia.adEngine.template.porvata', [
 	 * @param {object} params.height - Player height
 	 * @param {string} [params.onReady] - Callback executed once player is ready
 	 * @param {string} [params.vastUrl] - Vast URL (DFP URL with page level targeting will be used if not passed)
+	 * @param {integer} [params.vpaidMode] - VPAID mode from IMA: https://developers.google.com/interactive-media-ads/docs/sdks/html5/v3/apis#ima.ImaSdkSettings.VpaidMode
 	 */
 	function show(params) {
 		var settings = videoSettings.create(params);
 
 		log(['show', params], log.levels.debug, logGroup);
 
-		if (params.vpaidMode === googleIma.vpaidMode.INSECURE) {
-			params.originalContainer = params.container;
-			params.container = getVideoContainer(params.slotName);
-		}
-
 		hideOtherBidsForVeles(params);
 
 		porvata.inject(settings).then(function (video) {
-			if (params.vpaidMode === googleIma.vpaidMode.INSECURE) {
-				var videoPlayer = params.container.querySelector('.video-player');
+			var imaVpaidMode = win.google.ima.ImaSdkSettings.VpaidMode,
+				videoPlayer;
+
+			if (params.vpaidMode === imaVpaidMode.INSECURE) {
+				params.originalContainer = params.container;
+				params.container = getVideoContainer(params.slotName);
```

Comment:
Well, it's all wrapped in a condition which checks if it's `INSECURE`, so I'm sure it's not safe ;) Do you think moving it here is a big difference comparing to the previous place? I needed to move it here, so we're sure IMA is loaded and we can re-use IMA's enums. What other options do you think we have?